crates/fuel-core/src/service/adapters/p2p.rs

```diff
@@ -49,3 +55,17 @@ impl BlockHeightImporter for BlockImporterAdapter {
         )
     }
 }
+
+impl TxPool for TxPoolAdapter {
+    fn get_all_tx_ids(&self) -> Vec<TxId> {
+        self.service.get_all_tx_ids()
+    }
+
+    fn get_full_txs(&self, tx_ids: Vec<TxId>) -> Vec<Option<Transaction>> {
+        self.service
+            .find(tx_ids)
+            .into_iter()
+            .map(|tx_info| tx_info.map(|tx| tx.tx().as_ref().into()))
```

Comment:
I see. Yeah, we can solve it by using `Arc<Transaction>` on ports and adapter levels. Instead of using `Arc<PoolTransaction>` in the TxPool, we could use `struct PoolTransaction(Arc<Transaction>)`. Then, in this case, we could easily get `Arc<Transaction>`.

It is okay to do this change in a separate PR. It would be nice to make this change before merging this PR, if possible. But it is up to you=)

Reply:
I can change this in the transaction pool v2. Can we wait tx pool v2 before merging this ?

Reply:
I think it is better to do in a separate small PR. This branch and TxPoolV2 could use this branch as a base for this change

Reply:
After discussion I decided to go for a special type in P2P that allow to serialize a pool transaction or simple transaction as transaction and deserialize only in transactiton

Reply:
Here is a diff showing the changes that implement this : https://github.com/FuelLabs/fuel-core/compare/268d794...5e16d87